File: Programmers/Python/Level1/Ponketmon.py
def solution(nums):
    pon = list(set(nums))
    case = int(len(nums) / 2)
    
    if len(pon) >= case:
        return case
    else:
        return len(pon)

# 중복을 제거한 nums의 길이와 nums길이/2 중에서 최솟값을 return 
#def solution(nums):
#    return min(len(nums)/2, len(set(nums)))

File: Programmers/C++/Level1/Ponketmon.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int selection = nums.size() / 2;
    
    sort(nums.begin(), nums.end());
    nums.erase(unique(nums.begin(), nums.end()), nums.end());
    
    if(nums.size() >= selection) 
    {
        return selection;
    }
    else
    {
        return nums.size();
    }
}